airflow/models/dagrun.py

```diff
@@ -793,6 +793,8 @@ def get_previous_dagrun(
         :param session: SQLAlchemy ORM Session
         :param state: the dag run state
         """
+        if dag_run.logical_date is None:
```

Comment:
But is this correct for all manually triggered DAG runs when logical_date is None?